### PR TITLE
Drafting the project's Readme.md file and adding the infrastructure.md file

### PR DESCRIPTION
**Pull request by alkofahi**
* Copied the infrastructure.md file from the team's repo
* updated the README.md file
* Added/updated the README md files across all components

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -18,15 +18,20 @@ Guidance on onboarding samples to docs.microsoft.com/samples: https://review.doc
 Taxonomies for products and languages: https://review.docs.microsoft.com/new-hope/information-architecture/metadata/taxonomies?branch=master
 -->
 
-Give a short description for your sample here. What does it do and why is it important?
+This is a replica of the Urlist website but re-architected to use server based backend hosted in Azure AppService instead of being serverless. The original project can be found here https://github.com/burkeholland/the-urlist
+
+Please [check here](resources/infrastructure.md) to view and read about the new architecture.
```
Comment (wbreza) [inline README.md]: I would create a top level `docs` folder instead of `infrastructure`.  This is more common in git repos.

File: README.md [hunk 2]
```diff
@@ -35,19 +40,20 @@ Outline the file contents of the repository. It helps users navigate the codebas
...
+You need to have the following available
+
+* An Azure subscription
+* Check out the [API folder for backend prerequisites](api/README.md)
+* Check out the [Frontend folder for frontend prerequisites](frontend/README.md)
 
 ## Setup
 
-Explain how to prepare the sample once the user clones or downloads the repository. The section should outline every step necessary to install dependencies and set up any settings (for example, API keys and output folders).
+* You can check the deployment/deploy.sh script file to setup the Azure infrastructure automatically.
```
Comment (wbreza) [inline README.md]: I would link directly to the `deployment/docs/readme.md` file

File: README.md [hunk 3]
```diff
@@ -18,15 +18,21 @@ Guidance on onboarding samples to docs.microsoft.com/samples: https://review.doc
 Taxonomies for products and languages: https://review.docs.microsoft.com/new-hope/information-architecture/metadata/taxonomies?branch=master
 -->
 
-Give a short description for your sample here. What does it do and why is it important?
+This is a replica of the Urlist website but re-architected to use server based backend hosted in Azure AppService instead of being serverless. The original project repository can be found here https://github.com/the-urlist
```
Comment (Humoiz) [inline README.md]: Can we reword this "use server based backend" to something else?

File: docs/Infrastructure.md [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+Azure Frontdoor is used to provide a central frontdoor and mesh to all micro-services required for the Urlist application.  It consolidates all url endpoints to a single HTTP entrypoint.  Robust routing rules allow routing traffic to the correct microservice based on request paths.
+
+### Features
+- Intelligent routing based on geo-location and latency (edge delivery)
+- Single HTTP entry point (ex: www.urlist.com)
+- Handles TLS offloading for HTTPS (allows underlying services to use standard HTTP)
+- Caching for static content
+
+### Backend Pools
+Two backends are configured for the Urlist application. Backend pools can be configured with different priority or weights based on your business needs.  Each pool also must include a HTTP/HTTPS probe endpoint to monitor a regions health status.  Unhealthy nodes are automatically disabled from the pool.
```
Comment (Humoiz) [inline docs/Infrastructure.md]: We have more than 2 backends configured. I think we have configured it in westus, eastus and centralus.

File: pipelines/api-pipelines.yml [hunk 5]
```diff
@@ -133,7 +134,7 @@ stages:
               ConnectionType: 'AzureRM'
               azureSubscription: $(azureSubscription)
               appType: 'apiApp'
-              WebAppName: 'backend-$(businessUnit)-urlist-dev-$(region)'
```
Comment (Humoiz) [inline pipelines/api-pipelines.yml]: Thanks for making this change.

**Pull request merged**